kratos/utilities/function_parser_utility.cpp

```diff
@@ -141,8 +153,76 @@ void BasicGenericFunctionUtility::InitializeParser()
         const te_variable vars[] = {{"x", &x}, {"y", &y}, {"z", &z}, {"t", &t}, {"X", &X}, {"Y", &Y}, {"Z", &Z}};
 
         /* Compile the expression with variables. */
-        mpTinyExpr = te_compile(mFunctionBody.c_str(), vars, 7, &err);
-        KRATOS_ERROR_IF_NOT(mpTinyExpr) << "Parsing error in function: " << mFunctionBody << std::endl;
+        const bool c_like = StringUtilities::ContainsPartialString(mFunctionBody, "?") ? true : false;
+        const bool python_like = StringUtilities::ContainsPartialString(mFunctionBody, "if") ? true : false;
+        if (!(c_like || python_like)) {
+            mpTinyExpr[0] = te_compile(mFunctionBody.c_str(), vars, 7, &err);
+            KRATOS_ERROR_IF_NOT(mpTinyExpr[0]) << "Parsing error in function: " << mFunctionBody << std::endl;
+        } else { // Ternary operator
+            mpTinyExpr.resize(3, nullptr);
+            std::string condition, first_function, second_function;
+
+            // C like ternary operator
+            std::vector<std::string> splitted_string;
+            if (c_like) {
+                splitted_string = StringUtilities::SplitStringByDelimiter(mFunctionBody, '?');
+                KRATOS_ERROR_IF(splitted_string.size() > 2) << "Nested ternary functions not supported" << std::endl;
+                condition = splitted_string[0];
+                splitted_string = StringUtilities::SplitStringByDelimiter(splitted_string[1], ':');
+                first_function = splitted_string[0];
+                second_function = splitted_string[1];
+            } else { // Python like ternary operator
+                std::string aux_string = StringUtilities::ReplaceAllSubstrings(mFunctionBody, "if", "$");
+                KRATOS_ERROR_IF(splitted_string.size() > 2) << "Nested ternary functions not supported" << std::endl;
+                splitted_string = StringUtilities::SplitStringByDelimiter(aux_string, '$');
+                first_function = splitted_string[0];
+                aux_string = StringUtilities::ReplaceAllSubstrings(splitted_string[1], "else", "$");
+                splitted_string = StringUtilities::SplitStringByDelimiter(aux_string, '$');
+                condition = splitted_string[0];
+                second_function = splitted_string[1];
```

Comment:
We (@KratosMultiphysics/technical-committee) mean that in here you assume that the "else" is always present.

Reply:
Done